### PR TITLE
MSM: Unify FFA & test circuits to use generic interpreter/cap pattern

### PR DESCRIPTION
**Pull request by volhovm**
* Switches existing MSM circuits (FADD + Test) to the Cap/Interpreter pattern
* Removes copy-paste
* Adds tests where needed
* Adds lookups support

### CODE REVIEW

File: msm/src/ffa/lookups.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        match self {
+            Self::RangeCheck1BitSigned => {
+                value == F::zero() || value == F::one() || value == F::zero() - F::one()
+            }
+            Self::RangeCheck15 => value.to_biguint() < BigUint::from(2u128.pow(15)),
+        }
+    }
+}
+
+impl LookupTable {}
```
Comment (dannywillems) [inline msm/src/ffa/lookups.rs:87]: Required?

**Review by dannywillems — APPROVED** [review]
Love it. Not sure it is ready, but approving up to now :).

**Comment by volhovm** [comment]
@dannywillems thanks! This is ready now, the CI fails because of some stupid 404... Feel free to merge when it passes!

File: msm/src/ffa/lookups.rs (hunk 1 above)
Comment (volhovm) [reply msm/src/ffa/lookups.rs]: Thanks, fixed!

**Review by dannywillems — APPROVED** [review]

**Pull request merged**